### PR TITLE
add isLoading to useOrderbook, get symbol and interval as props in kl…

### PR DESCRIPTION
**Pull request by alirezahematidev**
…ine provider and hook

### CODE REVIEW

**Comment by alirezahematidev** [comment]
.

File: src/hitobit-hooks/src/ordersBook/index.tsx [hunk 1]
```diff
@@ -177,8 +179,10 @@ const OrderBookProvider = memo<Props>(({ children }) => {
     marketTicker?.lastMarketInfoChangeDate,
   ]);
 
+  const isLoading = isMarketTickerLoading || isDepthLoading;
+
   return (
-    <OrderBookContext.Provider value={state}>
+    <OrderBookContext.Provider value={{ ...state, isLoading }}>
```
Comment (hosseinmd) [inline src/hitobit-hooks/src/ordersBook/index.tsx:185]: Has performance issue.
When assigning object like this to value. This cause to re-render all consumers
Spread to new object in useMemo

File: src/hitobit-hooks/src/kline/index.tsx [hunk 2]
```diff
@@ -77,18 +76,15 @@ const KlinesProvider = ({ children }: Props) => {
...
+  symbol,
+  interval,
+}: {
+  symbol: string;
+  interval?: KLineInterval;
+}) {
   const { data: klines, isLoading: isKlinesLoading } =
     useGetExchangeV1PublicKlines(
-      { symbol, interval: constantInterval || interval, limit: 200 },
+      { symbol, interval: interval || "OneDay", limit: 200 },
```
Comment (hosseinmd) [inline src/hitobit-hooks/src/kline/index.tsx:87]: Interval has already defaulted value

Comment (hosseinmd) [reply src/hitobit-hooks/src/kline/index.tsx:87]: Don't need to add default value again

File: src/hitobit-hooks/src/kline/index.tsx [hunk 3]
```diff
@@ -77,18 +76,15 @@ const KlinesProvider = ({ children }: Props) => {
...
-    }
-  | undefined = {}) {
-  const { selectedSymbol: { symbol } = {}, interval } =
-    selectedSymbolStore.useState();
-
+  symbol,
+  interval,
+}: {
+  symbol: string;
+  interval?: KLineInterval;
```
Comment (hosseinmd) [inline src/hitobit-hooks/src/kline/index.tsx:83]: interval should have value, give default value or change to required

**Review by hosseinmd — CHANGES_REQUESTED** [review]

**Review by hosseinmd — APPROVED** [review]

**Pull request merged**